node/network_submodule.go

```diff
@@ -0,0 +1,24 @@
+package node
+
+import (
+	"github.com/filecoin-project/go-filecoin/net"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/routing"
+)
+
+// NetworkSubmodule enhances the `Node` with networking capabilities.
+type NetworkSubmodule struct {
+	NetworkName string
+
+	host host.Host
+
+	PeerHost host.Host
```

Comment:
Having 2 hosts fields seems weird to me. I understand its not the intent of this PR to address this so I have filed an issue to discuss it: https://github.com/filecoin-project/go-filecoin/issues/3477. Would you mind linking to it with a TODO?

Reply:
sure, linked